ScanUnsupportedChars/ScanUnsupportedChars.ps1

```diff
@@ -726,6 +744,30 @@ if ($FilesWithInvalidCharsFixedName.Count -gt 0)
 
         Write-Host "***************************Items failed to rename table end*****************************" -ForegroundColor Yellow
     }
+    elseif($PSBoundParameters.ContainsKey('DestinationPath'))
+    {
+        Write-Host "========================== File COPY/MOVE start ==========================================" -ForegroundColor Yellow
+        # Ensure UTF-8 Encoding
+        $PSDefaultParameterValues['*:Encoding'] = 'utf8'
+        
+        foreach ($file in $FilesWithInvalidCharsFixedName){
```

Comment:
Did you verify that it works? We added C# code in the script just to handle those special chars. Powershall does not handle all Unicode chars. Your logic to compute the destination path in the PS might lose the character fidelity.

I have some unsupported chars files for test purpose, shared here: \\SURENDEV\Tools\UnsupportedChars 

You can copy this share locally and experiment. This share contains unsupported chars files in single directory. It would be good to put them in directory hierarchy for testing.

Reply:
Couldn't find the Share mentioned but I've tested:

8967_1911_��提s+�F݄-Local.txt
s��本.pdf
😄﷯.txt

as File names. Though I checked with another Terminal with PowerShell 7 and recognized, that the active Code Page was not set to 65001 (unicode) so I've added both of them accordingly. As Start-Process opens another Shell and passes the Arguments to it, the Encoding was also necessary.